src/java/org/apache/cassandra/index/sai/disk/vector/BruteForceRowIdIterator.java

```diff
@@ -41,7 +44,7 @@
  * <p>
  * As an implementation detail, we use a PriorityQueue to maintain state rather than a List and sorting.
  */
-public class BruteForceRowIdIterator extends AbstractIterator<ScoredRowId>
+public class BruteForceRowIdIterator implements CloseableIterator<ScoredRowId>
```

Comment:
I still prefer extending AbstractIterator vs having next() rely on side effects in hasNext(), fwiw

Reply:
This must have been a minor misunderstanding from 929. Because we wanted to return `CloseableIterator`, I replaced `extend AbstractIterator` with `implements CloseableIterator` on several new iterators from 929. Those iterators all follow the pattern in this PR where `hasNext` is called in `next` block and has a side effect. I agree that it is simpler to extend `AbstractIterator`, and I just noticed that we have multiple `AbstractIterator` classes in the C* project. In `org.apache.cassandra.utils`, we have one that implements `CloseableIterator`, and in `org.apache.cassandra.index.sai.utils`, we have one that only implements `PeekingIterator`. Looking into using the one that already implements `CloseableIterator`.

Strictly speaking, both `AbstractIterator` implementations will call `hasNext()` if the next value has not been materialized already. However, I'd prefer to leave the logic to a single super class instead of re-implementing it in every iterator.